In [11]:
sales_data["Revenue"] = (
    sales_data["OrderItemQuantity"] * sales_data["PerUnitPrice"]
)

total_revenue = sales_data["Revenue"].sum()
total_orders = sales_data["OrderID"].nunique()
average_order_value = total_revenue / total_orders
total_units_sold = sales_data["OrderItemQuantity"].sum()

print("Total Revenue:", round(total_revenue, 2))
print("Total Orders:", total_orders)
print("Average Order Value:", round(average_order_value, 2))
print("Total Units Sold:", total_units_sold)

Total Revenue: 30520090.71
Total Orders: 400
Average Order Value: 76300.23
Total Units Sold: 35696


In [12]:
category_sales = (
    sales_data.groupby("CategoryName")["Revenue"]
    .sum()
    .reset_index()
    .sort_values("Revenue", ascending=False)
)

product_sales = (
    sales_data.groupby("ProductName")["Revenue"]
    .sum()
    .reset_index()
    .sort_values("Revenue", ascending=False)
)

print("Revenue by Category")
print(category_sales.to_string(index=False))

print("\nTop 10 Products by Revenue")
print(product_sales.head(10).to_string(index=False))

print("\nTop Category:", category_sales.iloc[0]["CategoryName"])
print("Best Product:", product_sales.iloc[0]["ProductName"])

Revenue by Category
CategoryName     Revenue
     Storage 10712932.17
  Video Card  5746481.52
Mother Board  5555189.14
         CPU  5366747.48
         RAM  3138740.40

Top 10 Products by Revenue
                      ProductName    Revenue
         G.Skill Ripjaws V Series 1063369.12
                G.Skill Trident Z  903422.14
             PNY SSD9SC240GMDA-RB  683691.83
       Corsair Dominator Platinum  634488.39
                Samsung MZ-V6E500  550487.00
          Crucial CT1050MX300SSD1  472802.75
Kingston Fury (HX316C10FB/8) 8 GB  389855.70
       Western Digital WD5000AACS  377972.32
           Crucial CT525MX300SSD1  357348.30
            Samsung MZ-75E120B/AM  344872.35

Top Category: Storage
Best Product: G.Skill Ripjaws V Series


In [13]:
sales_data["Month"] = sales_data["OrderDate"].dt.to_period("M").astype(str)

monthly_sales = (
    sales_data.groupby("Month")["Revenue"]
    .sum()
    .reset_index()
    .sort_values("Month")
)

print(monthly_sales.to_string(index=False))

  Month    Revenue
2013-06  422714.63
2015-04  378220.83
2015-05  565169.36
2015-06  435613.46
2015-10 1097945.92
2015-12  926421.98
2016-02 1402024.96
2016-05  700329.81
2016-06 2150980.94
2016-07  144094.47
2016-08 1328333.52
2016-09 1874581.89
2016-10 2208118.86
2016-11 1632880.33
2016-12 1597785.75
2017-01 1242340.30
2017-02 4573355.78
2017-03 2611744.27
2017-04  938281.17
2017-05  992528.03
2017-06  134097.68
2017-08 1368925.87
2017-09  950200.05
2017-10  450340.30
2017-11  393060.55


In [14]:
customer_sales = (
    sales_data.groupby(["CustomerID", "CustomerName"])
    .agg(
        Total_Revenue=("Revenue", "sum"),
        Total_Orders=("OrderID", "nunique"),
        Total_Units=("OrderItemQuantity", "sum")
    )
    .reset_index()
    .sort_values("Total_Revenue", ascending=False)
)

print("Top 10 Customers by Revenue")
print(customer_sales.head(10).to_string(index=False))

repeat_customers = (customer_sales["Total_Orders"] > 1).sum()

print("\nTotal Customers:", customer_sales["CustomerID"].nunique())
print("Customers with Repeat Orders:", repeat_customers)

Top 10 Customers by Revenue
 CustomerID       CustomerName  Total_Revenue  Total_Orders  Total_Units
       2659      Shields Seffi      600155.00             1          145
       2546         Bill Stein      550487.00             1          133
       2604        Daina Combs      389855.70             1          130
       2573          Mac Mckay      344872.35             1          115
       2567 Alessandra Estrada      328038.42             1          138
       2401     Samara Barnett      323400.00             1          147
       2607   Darron Robertson      307683.51             1          149
       2456     Glinda Lambert      302398.88             1          112
       2513      Herman Stokes      300139.18             1           94
       2410        Reva Fuller      292382.07             1          123

Total Customers: 400
Customers with Repeat Orders: 0


In [15]:
total_revenue = sales_data["Revenue"].sum()
total_orders = sales_data["OrderID"].nunique()
average_order_value = total_revenue / total_orders
total_units_sold = sales_data["OrderItemQuantity"].sum()

top_category = category_sales.iloc[0]["CategoryName"]
best_product = product_sales.iloc[0]["ProductName"]

kpi_summary = pd.DataFrame({
    "KPI": [
        "Total Revenue",
        "Total Orders",
        "Average Order Value",
        "Total Units Sold",
        "Top Category",
        "Best Product"
    ],
    "Value": [
        round(total_revenue, 2),
        total_orders,
        round(average_order_value, 2),
        total_units_sold,
        top_category,
        best_product
    ]
})

print(kpi_summary.to_string(index=False))

kpi_summary.to_csv(
    "/content/cleaned_data/kpi_summary.csv",
    index=False
)

                KPI                    Value
      Total Revenue              30520090.71
       Total Orders                      400
Average Order Value                 76300.23
   Total Units Sold                    35696
       Top Category                  Storage
       Best Product G.Skill Ripjaws V Series


In [16]:
highest_month = monthly_sales.loc[monthly_sales["Revenue"].idxmax()]
lowest_month = monthly_sales.loc[monthly_sales["Revenue"].idxmin()]

print("Highest Sales Month:", highest_month["Month"])
print("Revenue:", round(highest_month["Revenue"], 2))

print("\nLowest Sales Month:", lowest_month["Month"])
print("Revenue:", round(lowest_month["Revenue"], 2))

Highest Sales Month: 2017-02
Revenue: 4573355.78

Lowest Sales Month: 2017-06
Revenue: 134097.68



## Key Observations

1. Total recorded revenue is 3,052,000.71 across 400 orders.
2. The average order value is 7,630.23.
3. A total of 3,569 units were sold.
4. Storage was the category with the highest recorded revenue.
5. G.Skill Ripjaws V Series was the product with the highest recorded revenue.
6. February 2017 had the highest recorded monthly revenue of 457,335.78.
7. June 2017 had the lowest recorded monthly revenue of 134,097.68.
8. The analysis found 400 customers, with no repeat customers identified in the current order data.